### PR TITLE
feat: Delete data dump from S3 after data load has been successful

### PR DESCRIPTION
**Pull request by antroy-madetech**
Addresses [DBTP-1109](https://uktrade.atlassian.net/browse/DBTP-1109)

- Delete data dump from S3 after data load has been successful
- Delete the old database-copy Docker files
- Move the new Docker files from database-copy2 into database-copy 
- Added a `database copy` command that rolls the `dump` and `load` commands together

---
### Title:
- [ ] Scope included as per [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Ticket reference included (unless it's a quick out of ticket thing)
### Description:
- [ ] Link to ticket included (unless it's a quick out of ticket thing)
- [ ] Includes tests (or an explanation for why it doesn't)
- [ ] If the work includes user interface changes, before and after screenshots included in description
- [ ] Includes any applicable changes to the documentation in this code base
- [ ] Includes link(s) to any applicable changes to the documentation in the [DBT Platform Documentation](https://platform.readme.trade.gov.uk/) (can be to a pull request)
### Tasks:
- [ ] [Trigger the pull request regression tests for this branch](https://github.com/uktrade/platform-tools?tab=readme-ov-file#regression-tests) and confirm that they are passing

[DBTP-1109]: https://uktrade.atlassian.net/browse/DBTP-1109?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dbt_platform_helper/commands/database_helpers.py [hunk 1]
```diff
@@ -1,124 +1,205 @@
...
-    )
-
-    return response.get("tasks", [{}])[0].get("taskArn")
+class MaintenancePageProvider:
+    def __init__(self):
+        self.offline = offline_command
+        self.online = online_command
 
 
 class DatabaseCopy:
```
Comment (WillGibson) [inline dbt_platform_helper/commands/database_helpers.py:24]: How do you feel about moving the classes etc. into a new `src` or similar directory and starting to tidily namespace it all?

File: dbt_platform_helper/commands/database_helpers.py [hunk 2]
```diff
@@ -1,124 +1,205 @@
...
+        vpc_config_fn: Callable[[Session, str, str, str], Vpc] = get_vpc_info_by_name,
+        db_connection_string_fn: Callable[
+            [Session, str, str, str, Callable], str
+        ] = get_connection_string,
+        maintenance_page_provider: Callable[
+            [str, str, list[str], str, str], None
+        ] = MaintenancePageProvider(),
+        input_fn: Callable[[str], str] = click.prompt,
+        echo_fn: Callable[[str], str] = click.secho,
+        abort_fn: Callable[[str], None] = abort_with_error,
```
Comment (WillGibson) [inline dbt_platform_helper/commands/database_helpers.py:40]: I haven't really looked at this thoroughly, but this feels like a lot of things going into the constructor.

It makes me wonder whether the `DatabaseCopy` class is doing a bit much. Maybe we can separate out the copying work from the input/output/side effects?

File: dbt_platform_helper/commands/database_helpers.py (hunk 1 above)
Comment (antroy-madetech) [reply dbt_platform_helper/commands/database_helpers.py]: We've had similar thoughts. "commands" module will be where the thin click wrappers will live. "domain" module for the business logic and "providers" module for the common code that connects with things outside the domain logic (most of the "utils" module probably belongs in here)

Comment (antroy-madetech) [reply dbt_platform_helper/commands/database_helpers.py]: (I'll write this up in the confluence doc for people to comment on)

File: dbt_platform_helper/commands/database_helpers.py (hunk 2 above)
Comment (antroy-madetech) [reply dbt_platform_helper/commands/database_helpers.py]: I think some of this stuff can be tidied up further. For example the input/echo/abort are all UI elements that can probably be moved into a single UserInterfaceProvider or something. Perhaps in a tidy up step after we get this command out for FFT to use.

File: dbt_platform_helper/COMMANDS.md [hunk 3]
```diff
@@ -987,23 +991,20 @@ platform-helper database (dump|load)
...
 
-- `--account-id <text>`
-
 - `--app <text>`
-
-- `--env <text>`
-
+  - The application name. Required unless you are running the command from your deploy repo
+- `--from <text>`
+  - This is required unless you are running the command from your deploy repo
```
Comment (ksugden) [inline dbt_platform_helper/COMMANDS.md:1003]:
```suggestion
  - The environment you are dumping data from.  This is required unless you are running the command from your deploy repo
```

Comment (ksugden) [inline dbt_platform_helper/COMMANDS.md:1003]:
```suggestion
  - The environment data is being dumped from. Required unless you are running the command from your deploy repo
```

Comment (ksugden) [inline dbt_platform_helper/COMMANDS.md:1003]: The environment you are dumping data from. Required unless you are running the command from your deploy repo

File: dbt_platform_helper/COMMANDS.md [hunk 4]
```diff
@@ -1016,22 +1017,62 @@ platform-helper database dump --account-id <account_id> --app <application>
...
 ```
 
 ## Options
 
-- `--account-id <text>`
-
 - `--app <text>`
+  - The application name. Required unless you are running the command from your deploy repo
+- `--to <text>`
+  - The environment you are loading data into
```
Comment (ksugden) [inline dbt_platform_helper/COMMANDS.md:1030]: Are these mandatory parameters or are they only required if not in deploy repo?  (dump command above would suggest the latter)

File: dbt_platform_helper/commands/environment.py [hunk 5]
```diff
@@ -47,158 +39,17 @@ def environment():
...
-        )
-        raise click.Abort
-
-    except ListenerNotFoundError:
-        click.secho(
-            f"No HTTPS listener found for environment {env} in the application {app}.", fg="red"
-        )
-        raise click.Abort
+    maintenance_page = MaintenancePageProvider()
+    maintenance_page.offline(app, env, svc, template, vpc)
```
Comment (ksugden) [inline dbt_platform_helper/commands/environment.py:43]: Naming:
maybe 
```maintenance_page_provider = MaintenancePageProvider()
    maintenance_page_provider.set_application_offline(app, env, svc, template, vpc)
```

File: dbt_platform_helper/domain/database_copy.py [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+        if not vpc_name:
+            if not Path(PLATFORM_CONFIG_FILE).exists():
+                self.abort_fn(
+                    "You must either be in a deploy repo, or provide the vpc name option."
+                )
+            config = load_and_validate_platform_config(disable_aws_validation=True)
+            vpc_name = config.get("environments", {}).get(env, {}).get("vpc")
+        return vpc_name
+
+    def run_database_copy_task(
```
Comment (ksugden) [inline dbt_platform_helper/domain/database_copy.py:117]: thoughts for later - we should have an EcsTaskService for doing this sort of thing

File: images/tools/database-copy/entrypoint.sh [hunk 7]
```diff
@@ -1,30 +1,102 @@
...
+    echo ${SERVICE_NAME}
+    UPDATE_DATA=$(aws ecs update-service --cluster "${ECS_CLUSTER}" --service "${service}" --desired-count "${COUNT}")
+    handle_errors $? "Failed to update service ${SERVICE_NAME}"
+    echo "${UPDATE_DATA}" | jq -r '"  Desired Count: \(.service.desiredCount)\n  Running Count: \(.service.runningCount)"'
+  done
+
+  clean_up
+
+  echo "Stopping data load"
+fi
```
Comment (ksugden) [inline images/tools/database-copy/entrypoint.sh:102]: This bash script is quite unwieldy.  Could we separate it into scripts for the different 'steps'?  Maybe should consider something like https://aws.amazon.com/step-functions/features/?pg=ln&sec=hs in the future.

File: images/tools/database-copy2/Dockerfile [hunk 8]
```diff
@@ -1,11 +0,0 @@
-FROM postgres:16
```
Comment (ksugden) [inline images/tools/database-copy2/Dockerfile:1]: Are we cleaning up this directory?

File: images/tools/database-copy/entrypoint.sh (hunk 7 above)
Comment (antroy-madetech) [reply images/tools/database-copy/entrypoint.sh:102]: We'll look at this in the tidy up step.

**Pull request merged**